src/Esfa.Vacancy.Register.Application/Queries/SearchApprenticeshipVacancies/SearchApprenticeshipVacanciesQueryHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Vacancy.Register.Application.Interfaces;
+using Esfa.Vacancy.Register.Domain.Entities;
+using Esfa.Vacancy.Register.Domain.Repositories;
+using FluentValidation;
+using FluentValidation.Results;
+using MediatR;
+
+namespace Esfa.Vacancy.Register.Application.Queries.SearchApprenticeshipVacancies
+{
+    public class SearchApprenticeshipVacanciesQueryHandler : IAsyncRequestHandler<SearchApprenticeshipVacanciesRequest, SearchApprenticeshipVacanciesResponse>
+    {
+        private readonly IValidator<SearchApprenticeshipVacanciesRequest> _validator;
+        private readonly IVacancySearchService _vacancySearchService;
+        private readonly IStandardRepository _standardRepository;
+
+        public SearchApprenticeshipVacanciesQueryHandler(
+            IValidator<SearchApprenticeshipVacanciesRequest> validator,
+            IVacancySearchService vacancySearchService,
+            IStandardRepository standardRepository)
+        {
+            _validator = validator;
+            _vacancySearchService = vacancySearchService;
+            _standardRepository = standardRepository;
+        }
+
+        public async Task<SearchApprenticeshipVacanciesResponse> Handle(SearchApprenticeshipVacanciesRequest request)
+        {
+            var validationResult = _validator.Validate(request);
+
+            if (!validationResult.IsValid)
+                throw new ValidationException(validationResult.Errors);
+
+            var searchParameters = new VacancySearchParameters
+            {
+                StandardSectorCodes = await ValidateStandardCodes(request.StandardCodes.Select(int.Parse))
```

Comment:
Do you need to call Trim() on each element, in the case of elements being " 1" or "1 ". Or should I be  😶 ?

Reply:
Parse handles it.